code/game/objects/items/devices/traitordevices.dm

```diff
@@ -264,6 +264,31 @@
 	max_charges = 8
 	flawless = TRUE
 
+/obj/item/fireproofing_injector
+	desc = "It contains an alien nanoswarm created by the technomancers of boron. Through near sorcerous feats via use of nanomachines, it enables its user to become fully fireproof."
+	icon = 'icons/obj/hypo.dmi'
+	icon_state = "combat_hypo"
+	var/used = FALSE
+
+/obj/item/fireproofing_injector/attack_self(mob/living/user)
+	if(HAS_TRAIT(user, TRAIT_RESISTHEAT))
+		to_chat(user, "You are already fireproof!")
```

Comment:
The `to_chat`s in this proc could use some span classes, like `danger` or `notice`.